### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by DDCecilia**
Assignment 2

### CODE REVIEW

**Comment by DDCecilia** [comment]
hhh

File: Assignment 2.Rmd [hunk 1]
```diff
@@ -87,12 +94,53 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+EDGE.st <- EDGE %>% group_by(to) %>% summarize(sum(count))
+names(EDGE.st) <- c("id", "count")
+EDGE.tc <- EDGE %>% group_by(from) %>% summarise(sum(count))
+EDGE.tc
+names(EDGE.tc) <- c("id", "count")
+total <- bind_rows(EDGE.st, EDGE.tc)
```
Comment (charles-lang) [inline Assignment 2.Rmd]: Be careful you are binding variables of the same type (EG - `factor` to `factor)

File: Assignment 2.Rmd [hunk 2]
```diff
@@ -87,12 +94,53 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
...
+VERTEX2 <- full_join(VERTEX, total, by = "id")
+g2 <- graph.data.frame(EDGE, directed = TRUE, vertices = VERTEX2)
+plot(g2, layout = layout.fruchterman.reingold, vertex.color = VERTEX2$gender)
+```
```
Comment (charles-lang) [inline Assignment 2.Rmd]: It looks like you have created the values to size the vertices with successfully, but didn't employ them in your graph? The vertices should be different sizes (`vertex.size = VERTEX2$sum(count)`

File: Assignment 2.Rmd [hunk 3]
```diff
@@ -87,12 +94,53 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
...
+DC6 = DC5 %*% t(DC5)
+diag(DC6) <- NA
+g3 = graph.adjacency(DC6, mode = "undirected")
+plot(g3, layout=layout.fruchterman.reingold)
+
+
+which.max(degree(g3))
+which.max(betweenness(g3))
+```
+
```
Comment (charles-lang) [inline Assignment 2.Rmd]: Can you explain the difference between the two centrality measures?

File: Assignment 2.Rmd [hunk 4]
```diff
@@ -87,12 +94,53 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
...
+library(dplyr)
+library(tidyr)
+DC1 <- read.csv("hudk4050-classes.csv", header=TRUE)
+DC2 <- unite(DC1, "First.Name", "Last.Name", col = name, sep = " ")
+DC2
+DC3 <- DC2 %>% gather(classnum, classcode, `Class.1`, `Class.2`, `Class.3`, `Class.4`, `Class.5`, `Class.6`) %>% select(-c(classnum))
+DC3
+DC3$classcode = gsub(" ", "", DC3$classcode)
+
+DC3 <- DC3 %>% filter(classcode != "HUDK4050") %>% filter(name != "ZIMO CHEN") %>% filter(classcode != "")
```
Comment (charles-lang) [inline Assignment 2.Rmd]: Great work wrangling 👍

File: Assignment 2.Rmd [hunk 5]
```diff
@@ -87,12 +94,53 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
...
+DC3$Count = 1
+DC4 <- spread(DC3, key = classcode, value = Count)
+row.names(DC4) = DC4$name
+DC4$name <- NULL
+DC4 = ifelse(is.na(DC4), 0, 1)
+DC5 = as.matrix(DC4)
+DC6 = DC5 %*% t(DC5)
+diag(DC6) <- NA
+g3 = graph.adjacency(DC6, mode = "undirected")
+plot(g3, layout=layout.fruchterman.reingold)
```
Comment (charles-lang) [inline Assignment 2.Rmd]: You may want to resize the labels, they are obscuring your graph quite a bit.

**Pull request closed without merging**